exposed-tests/src/test/kotlin/org/jetbrains/exposed/sql/tests/shared/ddl/CreateIndexTests.kt

```diff
@@ -83,4 +82,68 @@ class CreateIndexTests : DatabaseTestsBase() {
             assertEquals(true, TestTable.exists())
         }
     }
+
+
+    @Test
+    fun `test partial index`() {
+        val partialIndexTable = object : IntIdTable("PartialIndexTableTest") {
+            val name = varchar("name", 50)
+            val value = integer("value")
+            val anotherName = integer("anotherName")
+            val anotherValue = integer("anotherValue")
+            val flag = bool("flag")
+            init {
+                index("flag_index", columns = arrayOf(flag, name)) {
+                    flag eq true
+                }
+                index(columns = arrayOf(value, name)) {
+                    (name eq "aaa") and (value greaterEq 6)
+                }
+                uniqueIndex(columns = arrayOf(anotherValue))
+            }
+        }
+
+        withDb(TestDB.POSTGRESQL) {
+            SchemaUtils.createMissingTablesAndColumns(partialIndexTable)
+            assertTrue(partialIndexTable.exists())
+
+            // check that indexes are created and contain the proper filtering conditions
+            exec(
+                """SELECT indexname AS INDEX_NAME,
+                   substring(indexdef, strpos(indexdef, ' WHERE ') + 7) AS FILTER_CONDITION
+                   FROM pg_indexes
+                   WHERE tablename='partialindextabletest' AND indexname != 'partialindextabletest_pkey'
+                """.trimIndent()
+            ) {
+                var totalIndexCount = 0
+                while (it.next()) {
+                    totalIndexCount += 1
+                    val filter = it.getString("FILTER_CONDITION")
+
+                    when (it.getString("INDEX_NAME")) {
+                        "partialindextabletest_value_name" -> assertEquals(filter, "(((name)::text = 'aaa'::text) AND (value >= 6))")
+                        "flag_index" -> assertEquals(filter, "(flag = true)")
+                        "partialindextabletest_anothervalue_unique" -> assertTrue(filter.startsWith(" UNIQUE INDEX "))
+                    }
+                }
+                kotlin.test.assertEquals(totalIndexCount, 3, "Indexes expected to be created")
+            }
+
+            fun List<Column<*>>.names(): Set<String> { return map { identity(it) }.toSet() }
+            fun getIndexes(): List<Index> {
+                db.dialect.resetCaches()
+                return currentDialect.existingIndices(partialIndexTable)[partialIndexTable].orEmpty()
+            }
+
+            val dropIndex = Index(columns = listOf(partialIndexTable.value, partialIndexTable.name), unique = false).dropStatement().first()
+            kotlin.test.assertTrue(dropIndex.startsWith("DROP INDEX "), "Unique partial index must be created and dropped as index")
+            val dropUniqConstraint = Index(columns = listOf(partialIndexTable.anotherValue), unique = true).dropStatement().first()
```

Comment:
nit (for consistency across the codebase):
```suggestion
            val dropUniqueConstraint = Index(columns = listOf(partialIndexTable.anotherValue), unique = true).dropStatement().first()
```